task.py

```diff
@@ -275,7 +276,32 @@ def get_namespace(self) -> str:
     def get_duration(self) -> int:
         return self.ts.cfg_descr.get_tft().duration
 
+    def _fetch_default_resource_name(self) -> Union[str, None]:
+        data = self.client.oc_get("network-attachment-definition")
+        resource_name = None
+
+        if data is not None:
+            try:
+                r = data["items"][0]["metadata"]["annotations"].get(
+                    "k8s.v1.cni.cncf.io/resourceName"
+                )
+                if isinstance(r, str) and r:
+                    resource_name = r
+            except Exception:
+                pass
```

Comment:
Why are we passing on Exception?

Reply:
We pass here because we don't need to anything to resource_name if there is an exception (resource name will be None and that is what we will return).

Reply:
Wait shouldn't this function use "secondary_network_nad" ?

Reply:
working on adding that!